DataFormats/Headers/include/Headers/DataHeader.h

```diff
@@ -262,7 +256,7 @@ struct Descriptor {
     assert(source != nullptr && (*source == 0 || (length >= 0 && length <= N)));
   }
 
-  bool operator==(const Descriptor& other) const {return itg == other.itg;}
+  bool operator==(const Descriptor& other) const {return std::memcmp(str,other.str,N)==0;}
```

Comment:
yes. Plus in the general case it is much easier to type. Also - in general it does not suffer from possible unaligned access issues.